src/parser.cc

```diff
@@ -2761,6 +2761,93 @@ void checkJointAxisExpressedInValues(const sdf::Root *_root, Errors &_errors)
   }
 }
 
+//////////////////////////////////////////////////
+template <typename TPtr>
+void checkScopedJointAxisMimicValues(
+    const TPtr _scope, const std::string &_scopeType, Errors &errors)
+{
+  const std::vector<std::string> followerAxisNames = {"axis", "axis2"};
+  for (uint64_t j = 0; j < _scope->JointCount(); ++j)
+  {
+    auto joint = _scope->JointByIndex(j);
+    for (uint64_t a = 0; a < 2; ++a)
+    {
+      auto axis = joint->Axis(a);
+      if (axis)
+      {
+        const std::string &followerAxis = followerAxisNames[a];
+        auto mimic = axis->Mimic();
+        if (mimic)
+        {
+          if (mimic->Joint() == joint->Name() &&
+              mimic->Axis() == followerAxis)
+          {
+            errors.push_back({ErrorCode::JOINT_AXIS_MIMIC_INVALID,
```

Comment:
Do we need this logic if it's checked in `sdf::Joint`?

Reply:
for users who call `Root::Load`, I agree it is redundant to have it both here and in `Joint::Load`. I know gz-sim sometimes constructs objects using `Joint::Load`, but I'm not sure exactly how that data flow works, so in my ignorance I figured it was safer to have it in both places. Let me know if I should remove the check from `Joint::Load` or from here

Reply:
I think `Joint::Load` will end up being called no matter what since the check in `parser.cc` is operating on `sdf::Joint` DOM objects, so it doesn't seem necessary to have the same logic repeated here.

Reply:
makes sense; I've removed it from parser.cc in https://github.com/gazebosim/sdformat/pull/1166/commits/1e4624c458c5b27487be6a7c0b3407532b37f5af